Rename IsLoggedIn to LoginStatus

pooneh-nb opened the pull request:
#### bd922faa2e53e03f4e462c63c795bb70ffb8c8f4
<pre>
Rename IsLoggedIn to LoginStatus
<a href="https://bugs.webkit.org/show_bug.cgi?id=275974">https://bugs.webkit.org/show_bug.cgi?id=275974</a>

Reviewed by Charlie Wolfe.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Headers.cmake:
* Source/WebCore/SmartPointerExpectations/NoUncountedMemberCheckerExpectations:
* Source/WebCore/SmartPointerExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/page/LoginStatus.cpp: Renamed from Source/WebCore/page/LoggedInStatus.cpp.
(WebCore::LoginStatus::create):
(WebCore::LoginStatus::LoginStatus):
(WebCore::LoginStatus::setTimeToLive):
(WebCore::LoginStatus::hasExpired const):
(WebCore::LoginStatus::expiry const):
* Source/WebCore/page/LoginStatus.h: Renamed from Source/WebCore/page/LoggedInStatus.h.
(WebCore::LoginStatus::registrableDomain const):
(WebCore::LoginStatus::username const):
(WebCore::LoginStatus::credentialTokenType const):
(WebCore::LoginStatus::authenticationType const):
(WebCore::LoginStatus::loggedInTime const):
* Source/WebCore/page/Navigator+LoginStatus.idl: Renamed from Source/WebCore/page/Navigator+IsLoggedIn.idl.
* Source/WebCore/page/NavigatorLoginStatus.cpp: Renamed from Source/WebCore/page/NavigatorIsLoggedIn.cpp.
(WebCore::NavigatorLoginStatus::from):
(WebCore::NavigatorLoginStatus::supplementName):
(WebCore::NavigatorLoginStatus::setLoggedIn):
(WebCore::NavigatorLoginStatus::setLoggedOut):
(WebCore::NavigatorLoginStatus::isLoggedIn):
* Source/WebCore/page/NavigatorLoginStatus.h: Renamed from Source/WebCore/page/NavigatorIsLoggedIn.h.
* Tools/DumpRenderTree/TestOptions.cpp:
(WTR::TestOptions::defaults):
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebCore/LoggedInStatus.cpp: Removed.
* Tools/TestWebKitAPI/Tests/WebCore/LoginStatus.cpp: Added.
(TestWebKitAPI::TEST(LoginStatus, DefaultExpiryWebAuthn)):
(TestWebKitAPI::TEST(LoginStatus, DefaultExpiryPasswordManager)):
(TestWebKitAPI::TEST(LoginStatus, DefaultExpiryUnmanaged)):
(TestWebKitAPI::TEST(LoginStatus, CustomExpiryBelowLong)):
(TestWebKitAPI::TEST(LoginStatus, CustomExpiryBelowShort)):
(TestWebKitAPI::TEST(LoginStatus, RenewedExpiry)):
(TestWebKitAPI::TEST(LoginStatus, InvalidUsernames)):
(TestWebKitAPI::TEST(LoginStatus, ClampedExpiryLong)):
(TestWebKitAPI::TEST(LoginStatus, ClampedExpiryShort)):

Canonical link: <a href="https://commits.webkit.org/280497@main">https://commits.webkit.org/280497@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/763f43e11e41806a9bc5d1329e96475d730a9cd9

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/56729 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/36057 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/9203 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/60344 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/7172 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/58856 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/43680 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/7362 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/45962 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/5038 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/58759 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/33891 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/48966 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/26819 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/30671 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/6297 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/6175 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/49821 "Built successfully and passed tests") | [⏳ 🧪 api-ios](https://ews-build.webkit.org/#/builders/API-Tests-iOS-Simulator-EWS "Waiting to run tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/6568 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/62027 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/55970 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/642 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/6669 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/53225 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/645 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/49021 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/53244 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/12560 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/553 "Passed tests") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/77731 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/31886 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/12880 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/32971 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/34056 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/32717 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on Source/WebCore/page/NavigatorLoginStatus.h:
```diff
@@ -31,24 +31,23 @@ namespace WebCore {
 class Navigator;
-
-class NavigatorIsLoggedIn final : public Supplement<Navigator> {
+class NavigatorLoginStatus final : public Supplement<Navigator> {
```
You'll need to reflect the name change in `Navigator+IsLoggedIn.idl`, too.

```
    ...
    ImplementedBy=NavigatorLoginStatus
] partial interface Navigator {
    ...
```

charliewolfe commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -3460,12 +3460,12 @@ IsFirstPartyWebsiteDataRemovalReproTestingEnabled:
       default: false
 
 # FIXME: The 'Is' prefix is inconsistent with most other preferences and should be removed.
```
I think this FIXME could be removed, since "Is" is no longer in the name.

charliewolfe commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -3460,12 +3460,12 @@ IsFirstPartyWebsiteDataRemovalReproTestingEnabled:
 
 # FIXME: The 'Is' prefix is inconsistent with most other preferences and should be removed.
-IsLoggedInAPIEnabled:
+LoginStatusAPIEnabled:
```
This file is sorted by name. Since this was renamed we need to move it to keep alphabetical order.

pooneh-nb replied:
Which file?

charliewolfe replied:
In this file (UnifiedWebPreferences.yaml). LoginStatusAPIEnabled likely needs to be moved below LockdownFontParserEnabled.

charliewolfe reviewed (commented):
Looks good! There is one more reference to `IsLoggedInAPIEnabled` in `Tools/TestOptions.cpp` that should be renamed too. Also, it looks like the API test filename (`Tools/TestWebKitAPI/Tests/WebCore/LoggedInStatus.cpp`) wasn't renamed, it probably should be.

johnwilander reviewed (changes requested):
Nice seeing this be changed! Long overdue.

From the change log:
```
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
```
Are these under version control? I'm always wary of anything derived sources.

Also, see my inline comment on human readable name.

johnwilander commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -3460,12 +3460,12 @@ IsFirstPartyWebsiteDataRemovalReproTestingEnabled:
   category: dom
-  humanReadableName: "IsLoggedIn web API"
-  humanReadableDescription: "Enable the proposed IsLoggedIn web API"
+  humanReadableName: "LoginStatus web API"
```
Let's use a space and call it Login Status API.

johnwilander commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -3460,12 +3460,12 @@ IsFirstPartyWebsiteDataRemovalReproTestingEnabled:
-  humanReadableName: "IsLoggedIn web API"
-  humanReadableDescription: "Enable the proposed IsLoggedIn web API"
+  humanReadableName: "LoginStatus web API"
+  humanReadableDescription: "Enable the proposed LoginStatus web API"
```
Ditto.

Commit:
Rename IsLoggedIn to LoginStatus
https://bugs.webkit.org/show_bug.cgi?id=275974

Reviewed by Charlie Wolfe.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Headers.cmake:
* Source/WebCore/SmartPointerExpectations/NoUncountedMemberCheckerExpectations:
* Source/WebCore/SmartPointerExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/page/LoginStatus.cpp: Renamed from Source/WebCore/page/LoggedInStatus.cpp.
(WebCore::LoginStatus::create):
(WebCore::LoginStatus::LoginStatus):
(WebCore::LoginStatus::setTimeToLive):
(WebCore::LoginStatus::hasExpired const):
(WebCore::LoginStatus::expiry const):
* Source/WebCore/page/LoginStatus.h: Renamed from Source/WebCore/page/LoggedInStatus.h.
(WebCore::LoginStatus::registrableDomain const):
(WebCore::LoginStatus::username const):
(WebCore::LoginStatus::credentialTokenType const):
(WebCore::LoginStatus::authenticationType const):
(WebCore::LoginStatus::loggedInTime const):
* Source/WebCore/page/Navigator+LoginStatus.idl: Renamed from Source/WebCore/page/Navigator+IsLoggedIn.idl.
* Source/WebCore/page/NavigatorLoginStatus.cpp: Renamed from Source/WebCore/page/NavigatorIsLoggedIn.cpp.
(WebCore::NavigatorLoginStatus::from):
(WebCore::NavigatorLoginStatus::supplementName):
(WebCore::NavigatorLoginStatus::setLoggedIn):
(WebCore::NavigatorLoginStatus::setLoggedOut):
(WebCore::NavigatorLoginStatus::isLoggedIn):
* Source/WebCore/page/NavigatorLoginStatus.h: Renamed from Source/WebCore/page/NavigatorIsLoggedIn.h.
* Tools/DumpRenderTree/TestOptions.cpp:
(WTR::TestOptions::defaults):
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebCore/LoggedInStatus.cpp: Removed.
* Tools/TestWebKitAPI/Tests/WebCore/LoginStatus.cpp: Added.
(TestWebKitAPI::TEST(LoginStatus, DefaultExpiryWebAuthn)):
(TestWebKitAPI::TEST(LoginStatus, DefaultExpiryPasswordManager)):
(TestWebKitAPI::TEST(LoginStatus, DefaultExpiryUnmanaged)):
(TestWebKitAPI::TEST(LoginStatus, CustomExpiryBelowLong)):
(TestWebKitAPI::TEST(LoginStatus, CustomExpiryBelowShort)):
(TestWebKitAPI::TEST(LoginStatus, RenewedExpiry)):
(TestWebKitAPI::TEST(LoginStatus, InvalidUsernames)):
(TestWebKitAPI::TEST(LoginStatus, ClampedExpiryLong)):
(TestWebKitAPI::TEST(LoginStatus, ClampedExpiryShort)):

Canonical link: https://commits.webkit.org/280497@main
